mirrord-layer/src/pod_api.rs

```diff
@@ -57,7 +58,7 @@ pub async fn create_agent(
     agent_namespace: &str,
     log_level: String,
     agent_image: String,
-) -> Portforwarder {
+) -> Result<Portforwarder> {
     let env_config = config::Config::init_from_env().unwrap();
     let client = if env_config.accept_invalid_certificates {
         let mut config = Config::infer().await.unwrap();
```

Comment:
For example here, I would suggest changing it to:

```rust
... = Config.infer().await.expect("Failed to infer config!");
```

Or some nicer message (I'm lacking in creativity here :P).